src/controllers/product.controller.js

```diff
@@ -0,0 +1,90 @@
+import jwt from "jsonwebtoken";
+import models from "../database/models";
+
+export const addProduct = async (req, res) => {
+  try {
+    const authHeader = req.headers.authorization;
+    if (!authHeader) {
+      return res.status(401).json({ message: "Authorization header missing" });
+    }
+    const token = authHeader.split(" ")[1];
+    const decodedToken = jwt.verify(token, process.env.SECRET_KEY);
+    const sellerId = decodedToken.userId;
+
+    const user = await models.User.findByPk(sellerId);
+    if (!user) {
+      return res.status(400).json({ message: "Seller not found" });
+    }
+    if (
+      sellerId !== decodedToken.userId ||
+      decodedToken.userRole !== "vendor"
+    ) {
+      return res.status(403).json({ message: "Unauthorized access" });
+    }
+
+    const {
+      name,
+      price,
+      quantity,
+      available,
+      category,
+      bonus,
+      expiryDate,
+      ec,
+    } = req.body;
+    const images = req.body.images || [];
+    const existingProduct = await models.Product.findOne({ where: { name } });
+    if (existingProduct) {
+      return res.status(409).json({
+        message: "Product already exists you can update that product instead",
+      });
+    }
+    const product = await models.Product.create({
+      userId: sellerId,
+      name,
+      price,
+      quantity,
+      available,
+      category,
+      bonus,
+      images,
+      expiryDate,
+      ec,
+    });
+    res.status(201).json(product);
+  } catch (error) {
+    res.status(400).json({ error });
+  }
+};
+
+export const deleteProduct = async (req, res) => {
+  try {
+    const authHeader = req.headers.authorization;
+    if (!authHeader) {
+      return res.status(401).json({ message: "Authorization header missing" });
+    }
+    const token = authHeader.split(" ")[1];
+    const decodedToken = jwt.verify(token, process.env.SECRET_KEY);
+    const sellerId = decodedToken.userId;
+
+    const user = await models.User.findByPk(sellerId);
+    if (!user) {
+      return res.status(400).json({ message: "Seller not found" });
+    }
+
+    const productId = req.params.id;
+    const product = await models.Product.findOne({
+      where: { id: productId, userId: sellerId },
+    });
+    if (!product) {
+      return res.status(404).json({ message: "Product not found" });
+    }
+    await product.destroy();
+    res.status(204).end();
+  } catch (error) {
+    res.status(400).json({ error });
+  }
```

Comment:
use status code 400 inside catch block is not defining exact error better to use 500 for internal sever error